### PR TITLE
[#50] Added rebar3 support

### PR DESCRIPTION
**Pull request by davecaos**
Added rebar3 support , The repositories erlang-repo and erlang-repo-fail are only dummy repos for testing.

### CODE REVIEW

File: test/xref_runner_SUITE.erl [hunk 1]
```diff
@@ -50,6 +52,47 @@ not_xref_register_himself(_Config) ->
...
+  Path = "../../test/examples/erlang-repo-fail",
+  ct:comment("It runs"),
+  {ok, OldCwd} = file:get_cwd(),
+  file:set_cwd(Path),
+  RepoName = "erlang-repo-fail",
+  rebar3_clean(RepoName),
+  rebar3_compile(RepoName), 
+  [Warning] = xref_runner:check(),
+
+  #{ line      := 9
```
Comment (gadgetci) [inline test/xref_runner_SUITE.erl]: According to **Elvis**:

> Line 79 has 1 trailing whitespace characters.

File: test/xref_runner_SUITE.erl [hunk 2]
```diff
@@ -50,6 +52,47 @@ not_xref_register_himself(_Config) ->
...
+  RepoName = "erlang-repo-fail",
+  rebar3_clean(RepoName),
+  rebar3_compile(RepoName), 
+  [Warning] = xref_runner:check(),
+
+  #{ line      := 9 
+   , check     := undefined_function_calls
+   , filename  := "test/examples/erlang-repo-fail/src/erlang_repo_sup.erl"
+   , source    := {erlang_repo_sup,init,1}
+   , target    := {erlang_repo_app,non_exist_function,0}
```
Comment (gadgetci) [inline test/xref_runner_SUITE.erl]: According to **Elvis**:

> Missing space right "," on line 83

File: test/xref_runner_SUITE.erl [hunk 3]
```diff
@@ -50,6 +52,47 @@ not_xref_register_himself(_Config) ->
...
+
+-spec check_rebar3_build_fail(config()) -> {comment, string()}.
+check_rebar3_build_fail(_Config) ->
+  Path = "../../test/examples/erlang-repo-fail",
+  ct:comment("It runs"),
+  {ok, OldCwd} = file:get_cwd(),
+  file:set_cwd(Path),
+  RepoName = "erlang-repo-fail",
+  rebar3_clean(RepoName),
+  rebar3_compile(RepoName),
```
Comment (gadgetci) [inline test/xref_runner_SUITE.erl]: According to **Elvis**:

> Line 76 has 1 trailing whitespace characters.

File: test/xref_runner_SUITE.erl [hunk 4]
```diff
@@ -50,6 +52,47 @@ not_xref_register_himself(_Config) ->
...
+  file:set_cwd(Path),
+  RepoName = "erlang-repo-fail",
+  rebar3_clean(RepoName),
+  rebar3_compile(RepoName), 
+  [Warning] = xref_runner:check(),
+
+  #{ line      := 9 
+   , check     := undefined_function_calls
+   , filename  := "test/examples/erlang-repo-fail/src/erlang_repo_sup.erl"
+   , source    := {erlang_repo_sup,init,1}
```
Comment (gadgetci) [inline test/xref_runner_SUITE.erl]: According to **Elvis**:

> Missing space right "," on line 82

File: test/xref_runner_SUITE.erl (hunk 2 above)
Comment (gadgetci) [inline test/xref_runner_SUITE.erl]: According to **Elvis**:

> Missing space right "," on line 83

File: test/xref_runner_SUITE.erl (hunk 4 above)
Comment (gadgetci) [inline test/xref_runner_SUITE.erl]: According to **Elvis**:

> Missing space right "," on line 82

File: src/xref_runner.erl [hunk 5]
```diff
@@ -100,13 +100,33 @@ default_checks() ->
...
 ebin() ->
+  hd([Path || Path <- [rebar3_path(), erlangmk_path(), default_path()],
+              Path =/= undefined]).
+
+rebar3_path() ->
+  case filelib:is_dir("_build/default/lib") of
+    true  ->
+      Dir = filename:absname("_build/default/lib"),
+      AppName = atom_to_list(current_app_name()),
+      [Dir, "/" , AppName, "/ebin/"];
```
Comment (elbrujohalcon) [inline src/xref_runner.erl]: use `filename:join/3`

File: src/xref_runner.erl [hunk 6]
```diff
@@ -100,13 +100,33 @@ default_checks() ->
...
-    true -> filename:absname("ebin");
-    false -> filename:absname(".")
+    true  -> filename:absname("ebin");
+    false -> undefined
   end.
 
+default_path() -> ".".
+
+current_app_name() ->
+  [File] = filelib:wildcard("src/*.app.src"),
```
Comment (elbrujohalcon) [inline src/xref_runner.erl]: rebar3 supports both src/_.app.src and ebin/_.app, we should also support both.

**Comment by elbrujohalcon** [comment]
You should add `.rebar3` and `_build` to `.gitignore` and remove those folders from git.
Maybe you should add `.gitignore`s to the example sub-folders.

File: src/xref_runner.erl [hunk 7]
```diff
@@ -100,13 +100,33 @@ default_checks() ->
   , locals_not_used
   , deprecated_function_calls
   ].
-
+-spec ebin() -> any().
```
Comment (elbrujohalcon) [inline src/xref_runner.erl]: It's not `any()`, the result is a `file:name_all()`

File: test/xref_runner_SUITE.erl [hunk 8]
```diff
@@ -50,6 +52,47 @@ not_xref_register_himself(_Config) ->
...
+  file:set_cwd(Path),
+  RepoName = "erlang-repo-fail",
+  rebar3_clean(RepoName),
+  rebar3_compile(RepoName),
+  [Warning] = xref_runner:check(),
+
+  #{ line      := 9
+   , check     := undefined_function_calls
+   , filename  := "test/examples/erlang-repo-fail/src/erlang_repo_sup.erl"
+   , source    := {erlang_repo_sup,init, 1}
```
Comment (gadgetci) [inline test/xref_runner_SUITE.erl]: According to **Elvis**:

> Missing space right "," on line 82

File: test/xref_runner_SUITE.erl [hunk 9]
```diff
@@ -50,6 +52,47 @@ not_xref_register_himself(_Config) ->
...
+   , target    := {erlang_repo_app,non_exist_function, 0}
```
Comment (gadgetci) [inline test/xref_runner_SUITE.erl]: According to **Elvis**:

> Missing space right "," on line 83

File: src/xref_runner.erl [hunk 10]
```diff
@@ -100,13 +100,33 @@ default_checks() ->
...
 ebin() ->
+  hd([Path || Path <- [rebar3_path(), erlangmk_path(), default_path()],
+              Path =/= undefined]).
+
+rebar3_path() ->
+  case filelib:is_dir("_build/default/lib") of
+    true  ->
+      Dir = filename:absname("_build/default/lib"),
+      AppName = atom_to_list(current_app_name()),
+      filename:join([Dir,AppName, "ebin"]);
```
Comment (gadgetci) [inline src/xref_runner.erl]: According to **Elvis**:

> Missing space right "," on line 113

**Pull request merged**